### PR TITLE
feat: add robustscaler

### LINKED ISSUE #650: Add `RobustScaler`
### Is your feature request related to a problem?

The two scalers, range scaler and standard scaler, cannot deal with outliers well.

### Desired solution

Add a new class `RobustScaler` similar to the [scikit-learn transformer with the same name](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.RobustScaler.html#sklearn.preprocessing.RobustScaler). It should be implemented using polars directly to avoid unnecessary conversions.

* Superclass: `InvertibleTableTransformer`
* Constructor parameters:
  * `column_names: str | list[str] | None = None` (keyword-only). List of columns to transform, if None all numeric columns passed in `fit`.
* Attributes:
  * `self._data_median: pl.DataFrame | None = None`
  * `self._data_scale: pl.DataFrame | None = None`
* `fit`:
  * Call `_check_columns_exist` to ensure columns to transform exist
  * Call `_check_columns_are_numeric` to ensure that columns to transform are numeric
  * Raise value error if `row_count` is 0
  * Create a new instance of the `RobustScaler`, don't mutate it in place
  * Compute the median (second quartile) for each column to transform and store it in `_data_median` of the copied transformer
  * Compute the [first and third quartile](https://en.wikipedia.org/wiki/Quartile) for each column to transform and store the difference (third - first) in `_data_scale` of the copied transformer
* `transform`:
  * `TransformerNotFittedError` if transformer is not fitted
  * Call `_check_columns_exist` to ensure columns to transform exist
  * Call `_check_columns_are_numeric` to ensure that columns to transform are numeric
  1. Subtract `_data_median` for each column to transform
  2. Divide result by `_data_scale` for each column to transform
* `inverse_transform`:
  * `TransformerNotFittedError` if transformer is not fitted
  * Call `_check_columns_exist` to ensure columns to transform exist
  * Call `_check_columns_are_numeric` to ensure that columns to transform are numeric
  1. Multiply by `_data_scale` for each column to transform
  2. Add `_data_median` for each column to transform

See implementation of `RangeScaler` and `StandardScaler` for implementation of similar functionality.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by Tarmandan**
Closes #650 

### Summary of Changes

Adds a RobustScaler class that works like the StandardScaler but uses median instead of mean and interquartile range instead of standard deviation. If the interquartile range is 0 it will only substract the median from all rows.

For now cannot handle columns containing NaN-values. See Issue #873

### CODE REVIEW

File: src/safeds/data/tabular/transformation/_robust_scaler.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Learning and transformation
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def fit(self, table: Table) -> RobustScaler:
+        """
+        Learn a transformation for a set of columns in a table.
+
+        This transformer is not modified.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/transformation/_robust_scaler.py:60]:
```suggestion
        **Note:** This transformer is not modified.
```

For consistency with more "modern" implementations of transformers. We'll update the others later.

File: src/safeds/data/tabular/transformation/_robust_scaler.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        result._data_median = _data_median
+        result._data_scale = _data_scale
+
+        return result
+
+    def transform(self, table: Table) -> Table:
+        """
+        Apply the learned transformation to a table.
+
+        The table is not modified.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/transformation/_robust_scaler.py:116]:
```suggestion
        **Note:** The given table is not modified.
```

File: src/safeds/data/tabular/transformation/_robust_scaler.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+        return Table._from_polars_lazy_frame(
+            table._lazy_frame.with_columns(columns),
+        )
+
+    def inverse_transform(self, transformed_table: Table) -> Table:
+        """
+        Undo the learned transformation.
+
+        The table is not modified.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/transformation/_robust_scaler.py:159]:
```suggestion
        **Note:** The given table is not modified.
```

**Review by lars-reimann — CHANGES_REQUESTED** [review]
Just a few nitpicks.

File: src/safeds/data/tabular/transformation/_robust_scaler.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            column_names = [name for name in table.column_names if table.get_column_type(name).is_numeric]
+        else:
+            column_names = self._column_names
+            _check_columns_exist(table, column_names)
+            _check_columns_are_numeric(table, column_names, operation="fit a RobustScaler")
+
+        if table.row_count == 0:
+            raise ValueError("The RobustScaler cannot be fitted because the table contains 0 rows")
+
+        # n-tiles = [0.25, 0.125, 0.0675]
```
[the remark this answers is not in the document]
Comment (Tarmandan) [reply src/safeds/data/tabular/transformation/_robust_scaler.py]: This was a holdover from when we were considering more granular handling of high stability. The comment has been removed

**Review by lars-reimann — APPROVED** [review]
Thanks, great work.

**Pull request merged**